### PR TITLE
Add feature weights support

### PR DESCRIPTION
**Pull request by mozjay0619**
This PR adds support for feature_weights parameter, which is xgb.DMatrix parameter since Xgboost version 1.3.0. This PR does not add this support for RayDeviceQuantileDMatrix class.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 setup(
     name="xgboost_ray",
     packages=find_packages(where=".", include="xgboost_ray*"),
-    version="0.1.14",
+    version="0.1.15",
```
Comment (Yard1) [inline setup.py:6]: There is no need to change the version in this PR.

**Review by Yard1 — COMMENTED** [review]
Thanks for the PR! Can you run `format.sh` in the root folder to make sure the code is properly formatted?

File: setup.py (hunk 1 above)
Comment (mozjay0619) [reply setup.py:6]: Yeah, thanks for the comment. Will do!

**Comment by mozjay0619** [comment]
@Yard1 I'm not sure how to proceed with the latest pytest failure. May be this is a temporary issue on Azure repo side? The error message says ``Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?``, but I am not sure how to do this. Could you please advise? Thanks!

**Comment by Yard1** [comment]
@mozjay0619 I just reran the job and it passed that point. It looks like a random failure not related to your changes.

**Comment by Yard1** [comment]
hmm @mozjay0619 we seem to have some actual errors, can you check?

**Comment by mozjay0619** [comment]
@Yard1 Sorry, let me explain the nature of the error since we've had this multiple times. The feature_weights leverages the colsample feature, so it is inherently stochastic. The weights are configured to be monotonically increasing with the last (9th) feature having the biggest contribution. But due to the stochasticity, the second to last feature sometimes overpowers the last feature. I increased the boosting rounds for convergence and lowered the colsample rates as well. And this time, I tested it against 100 random seed values. I can take more drastic measures but I don't want to alter the code too much from the original demo code provided by xgboost official doc...

**Comment by Yard1** [comment]
@mozjay0619 can we also set the seed as a parameter for xgboost? The numpy seed will not be carried over to the xgboost-ray Actors carrying out the training.

**Comment by mozjay0619** [comment]
@Yard1 Yeah, I had entertained that exact idea. But I couldn't find an easy way to do it since the xgboost.train() method does not have seed parameter, and it relies on numpy random state. I suspect this is why the demo code sets numpy random state. One hacky way might be to create a seed parameter to ``train`` method in xgboost_ray and use that number to set the local seed for each Actor? But I'm not sure what this would actually do in the context of distributed xgboost training...

**Comment by Yard1** [comment]
I believe it has a `seed` learning parameter but if the test passes here then I am fine with that. Thanks for iterating on this!

**Review by Yard1 — APPROVED** [review]

**Pull request merged**